### DISCUSSION TITLE
How to set email as userId in JPA Spring

### DISCUSSION THREAD

**Discussion by Shakhinah** [containers/podman#14739] [General]
GeneratedValue(strategy = GenerationType.IDENTITY)to set userId  usebut my coworker suggest to change it as emails so that easy to put in database. I do some research on how to set userId as email but I cannot find it. Can anyone help me? thanks

package com.disb.TicketingSystem.user;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Table(name = "users")
@Entity
public class User {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "user_id", nullable = false, updatable = false)
    private Long userId;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "email", nullable = false, length = 30)
    private String email;

    @Column(name = "password", nullable = false, length = 30)
    private String password;

    @Column(name = "user_type", nullable = false, length = 30)
    private String role;

    public User() {

    }

    public User(Long userId, String name, String email, String password, String role) {
        this.userId = userId;
        this.name = name;
        this.email = email;
        this.password = password;
        this.role = role;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }

}

**Comment by Luap99** [comment]
Did you create this discussion at the wrong repo? We have nothing to do with java.

**Reply by Shakhinah** [reply]
ohh okay .thank you for your respond
